webpack/assets/javascripts/react_app/components/common/Timer.js

```diff
@@ -0,0 +1,20 @@
+class Timer {
```

Comment:
Timer wraps the native setTimeout so yes it could be used. I think this is cleaner and reusable.